hardinfo/dt_util.c

```diff
@@ -895,6 +928,86 @@ int dtr_inh_find(dtr_obj *obj, char *qprop, int limit) {
     return ret;
 }
 
+dt_opp_range *dtr_get_opp_range(dtr *s, const char *name) {
+    dt_opp_range *ret = NULL;
+    dtr_obj *obj = NULL, *table_obj = NULL, *row_obj = NULL;
+    uint32_t opp_ph = 0;
+    const char *opp_table_path = NULL;
+    char *tab_compat = NULL, *tab_status = NULL;
+    const gchar *fn;
+    gchar *full_path;
+    GDir *dir;
+    uint64_t khz = 0;
+    uint32_t lns = 0;
+    char *row_status = NULL;
+
+    if (!s)
+        return NULL;
+
+    obj = dtr_obj_read(s, name);
+    if (!obj)
+        goto get_opp_finish;
+
+    opp_ph = dtr_get_prop_u32(s, obj, "operating-points-v2");
+    if (!opp_ph)
+        goto get_opp_finish;
+
+    opp_table_path = dtr_phandle_lookup(s, opp_ph);
+    if (!opp_table_path)
+        goto get_opp_finish;
+
+    table_obj = dtr_obj_read(s, opp_table_path);
+    if (!table_obj)
+        goto get_opp_finish;
+
+    tab_compat = dtr_get_prop_str(s, table_obj, "compatible");
+    tab_status = dtr_get_prop_str(s, table_obj, "status");
+
+    if (!tab_compat || strcmp(tab_compat, "operating-points-v2") != 0)
+        goto get_opp_finish;
+    if (tab_status && strcmp(tab_status, "disabled") == 0)
+        goto get_opp_finish;
+
+    ret = malloc(sizeof(dt_opp_range));
```

Comment:
It's usually not a good idea to mix malloc and g_new in a Glib program; it's too easy to forget to call free() instead of g_free()